### ISSUE TITLE
support detection or install node

### ISSUE THREAD

**Issue by xenoterracide** [node-gradle/gradle-node-plugin#183]
so gradle now supports configuring java like this, which will install a gradle version of java if the required version isn't already available. I think it would be cool to do that with node.

```
java {
  toolchain {
    languageVersion.set(JavaLanguageVersion.of(11))
  }
}
```

**Comment by deepy** [comment]
I think this functionality is Gradle internal and only supports the java toolchain, but ideally the same functionality should be exposed as an API so that we can just plug ourselves in there as a provider and get Gradle's help in figuring things out.

Or maybe we can re-implement their wheel, but yeah, this would be super-cool to get working

**Comment by xenoterracide** [comment]
yeah, I'm not certain that you can use their stuff, but you could do something like it probably. I would guess it's just a matter of searching the `PATH` seeing if the binary with the right version exists, and if not proceed to download. But maybe they have some internal api's that would make it easier for an arbitrary binary.

**Comment by deepy** [comment]
This may get split into multiple tasks, but for 3.3 I'll try getting the detection working

**Comment by birnbuazn** [comment]
Any progress on this? this would really help, because we already have node/npm installed in our CI/CD on Github, but on local developer machines, node should be downloaded automatically, if it's not there yet.

**Comment by deepy** [comment]
There's some work ongoing but so far everything produces unwieldy side-effects
But your use-case has an easy solution, set `download` to false when on CI

**Comment by birnbuazn** [comment]
> But your use-case has an easy solution, set download to false when on CI

That's exactly what I went forward with, yes.
